pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
```

Comment:
I don't think this should be labeled as specific to ingress.  I would define this as:

`SecretTypeTLS contains information about a TLS client or server secret. It is primarily used with TLS termination of the Ingress resource, but may be used in other types.`